src/webview/MessageList.js

```diff
@@ -335,6 +335,7 @@ type OuterProps = {|
   messages: $ReadOnlyArray<Message | Outbox>,
   initialScrollMessageId: number | null,
   showMessagePlaceholders: boolean,
+  startEditMessage: (editMessage: EditMessage) => void,
```

Comment:
MessageList types: Copy `startEditMessage` to `OuterProps`, where it belongs.
    
    Now, `OuterProps` matches the subset of `Props` that callers are
    meant to pass. If `OuterProps` must exist, this is what it should
    look like.

Ah, good catch, thanks.

Indeed this type definition ideally shouldn't need to exist. The reason for it lives in one of those commit messages where I probably should have put some of the information in a code comment:

commit 3168ff5c1b36cb03947bd9e431de98a266fdcfef
Author: Greg Price <greg@zulip.com>
Date:   Wed Sep 5 18:13:37 2018 -0700

    message list: Give a real type to the exported MessageList.
    
    By annotating the argument to `connect`, we enable Flow to infer the
    type of its return value -- aka `MessageList`, as seen by the rest of
    the code -- and type-check call sites.
    
    The repetition is unfortunate and I'd be glad to find a way to avoid
    it.  I tried `Props = OuterProps & { /* ... */ }`, but unfortunately
    that left Flow still thinking that `anchor`, `fetching`, etc. could be
    missing because they could in `OuterProps`.  This is much better than
    not having types, though.
    
    Also, because `connectActionSheet` lacks a type annotation, the
    relationship *between* `OuterProps` and `Props`, and the code that
    (*ahem*) connects them, isn't checked.  That'd be good to fix.
    
    We abuse this `Props` type in a lot of places where we really mean
    some other type that merely overlaps, and that contributes to some
    mysteries still to be resolved here.  I cleared up one --
    `highlightUnreadMessages` just never appears in something claiming to
    use this type -- but a few others are marked with comments.

I think the key point there is that `connectActionSheet` didn't have a type. (The `Props = OuterProps & /* … */` thing was an attempt to use [intersection types](https://flow.org/en/docs/types/intersections/); if I were attempting the same thing now, I'd write `Props = {| ...OuterProps, /* … */ |}` and work from there. Intersection types are potentially helpful for inexact object types -- which we were using then because we didn't entirely know better -- but don't have a lot of other applications.)

With the next commit in this PR, though, it does! So maybe we can drop `OuterProps` after that?

Or, given that `OuterProps` comes in independently handy in the last commit of the branch: maybe define it at the top and save the duplication by spreading it into `Props`? Just like we do `SelectorProps`, basically.

Reply:
> The reason for it lives in one of those commit messages where I probably should have put some of the information in a code comment

IIRC, I found this commit message, noted that it said `OuterProps` was necessary, then also realized that it's worth keeping for the last commit in the branch. I elided all that with just "If `OuterProps` must exist", probably because I didn't feel like explaining everything? 😛